src/braket/aws/aws_quantum_task.py

```diff
@@ -272,6 +272,8 @@ def result(self) -> Union[GateModelQuantumTaskResult, AnnealingQuantumTaskResult
         """
         if self._result or self._status(True) in self.NO_RESULT_TERMINAL_STATES:
             return self._result
+        if self._status(True) in self.RESULTS_READY_STATES:
```

Comment:
This will call get_quantum_task if we've never polled before. Is this intended? 
I was trying to fix this issue two lines above in https://github.com/aws/amazon-braket-sdk-python/pull/226/files

Reply:
I think that is reasonable, since there is no need for polling if a task is already done. Looks like we need this call to check status.

Reply:
The issue with this currently is if we've never polled before (self._metadata is empty), then this will poll immediately to full self._metadata and then possibly proceed to the async event loop, where it could poll immediately again. That would be an extra poll.